controllers/helpers.go

```diff
@@ -158,88 +158,102 @@ func GetCloudProviderSecret(d azure.ClusterDescriber, namespace, name string, ow
 		},
 	}
 
-	var data []byte
-	var err error
+	var controlPlaneConfig, workerNodeConfig *CloudProviderConfig
 
 	switch identityType {
 	case infrav1.VMIdentitySystemAssigned:
-		data, err = systemAssignedIdentityCloudProviderConfig(d)
-		if err != nil {
-			return nil, err
-		}
+		controlPlaneConfig, workerNodeConfig = systemAssignedIdentityCloudProviderConfig(d)
 	case infrav1.VMIdentityUserAssigned:
 		if len(userIdentityID) < 1 {
 			return nil, errors.New("expected a non-empty userIdentityID")
 		}
-		data, err = userAssignedIdentityCloudProviderConfig(d, userIdentityID)
-		if err != nil {
-			return nil, err
-		}
+		controlPlaneConfig, workerNodeConfig = userAssignedIdentityCloudProviderConfig(d, userIdentityID)
 	case infrav1.VMIdentityNone:
-		data, err = servicePrincipalCloudProviderConfig(d)
-		if err != nil {
-			return nil, err
-		}
+		controlPlaneConfig, workerNodeConfig = newCloudProviderConfig(d)
+	}
+
+	controlPlaneData, err := json.MarshalIndent(controlPlaneConfig, "", "    ")
+	if err != nil {
+		return nil, err
```

Comment:
Can we wrap this error so we can have a better idea of where this happened? We will see very similar errors from both control plane and worker node in the logs if we don't wrap.

Reply:
good call. updated